src/main/java/org/jenkinsci/plugins/github/util/XSSApi.java

```diff
@@ -0,0 +1,27 @@
+package org.jenkinsci.plugins.github.util;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * @author lanwen (Merkushev Kirill)
+ */
+public final class XSSApi {
+    private XSSApi() {
+    }
+
+    /**
+     * Method to filter invalid url for XSS. This url can be inserted to href safely
+     *
+     * @param urlString unsafe url
+     *
+     * @return safe url
+     */
+    public static String asValidHref(String urlString) {
+        try {
+            return new URL(urlString).toExternalForm();
```

Comment:
Expecting github url is **scheme + fqdn + project path** all other urls will be filtered